pilopt/src/lib.rs

```diff
@@ -120,31 +121,47 @@ fn remove_unreferenced_definitions<T: FieldElement>(pil_file: &mut Analyzed<T>)
     pil_file.remove_trait_impls(&impls_to_remove);
 }
 
-/// Builds a lookup-table that can be used to turn array elements
-/// (in form of their poly ids) into the names of the arrays.
+/// Builds a lookup-table that can be used to turn all symbols
+/// (including array elements) in the form of their poly ids, into the names of the symbols.
+/// The boolean flag indicates whether the symbol belongs to an array or not
```

Comment:
Would it makes sense to return `Option<usize>` instead and return the array index?